component/src/main/ballerina/ballerina/net/soap/natives.bal

```diff
@@ -81,80 +81,105 @@ public struct SoapError {
-        if (soapVersion == SoapVersion.SOAP11) {
-            req.setHeader("Content-Type", "text/xml");
-            req.addHeader("SOAPAction", request.soapAction);
-        } else {
-            req.setHeader("Content-Type", "application/soap+xml");
-        }
+
         Response soapResponse = {};
-        SoapError soapError = {};
-        resp, httpError = httpConnector.post("", req);
+        resp, httpError = httpConnector.post(path, req);
         if (resp != null) {
-            soapResponse = createResponse(resp, soapVersion);
-        }
-        if (httpError != null) {
-            soapError.msg = httpError.msg;
-            soapError.cause = httpError.cause;
-            soapError.stackTrace = httpError.stackTrace;
-            soapError.errorCode = httpError.statusCode;
+            soapResponse = createSOAPResponse(resp, soapVersion);
         }
-        return soapResponse, soapError;
+        return soapResponse, getSoapError(httpError);
     }
 
 }
 
-function getConnectorConfigs() (http:Options) {
-    http:Options option = {
-                              ssl: {
-                                       trustStoreFile:"${ballerina.home}/bre/security/client-truststore.jks",
-                                       trustStorePassword:"wso2carbon"
-                                   },
-                              followRedirects: {}
-                          };
-    return option;
+@Description { value:"Returns a SoapError from the HttpConnectorError" }
+@Param { value:"httpError: The HttpConnectorError" }
+@Return { value:"SoapError: The SoapError obtained from the HttpConnectorError" }
+function getSoapError(http:HttpConnectorError httpError) (SoapError){
+    SoapError soapError = {};
+    if (httpError != null) {
+        soapError.msg = httpError.msg;
+        soapError.cause = httpError.cause;
+        soapError.stackTrace = httpError.stackTrace;
+        soapError.errorCode = httpError.statusCode;
+    }
+    return soapError;
 }
 
-function createResponse (http:Response resp, SoapVersion soapVersion) (Response) {
+@Description { value:"Prepare a SOAP envelope with the xml to be sent." }
+@Param { value:"request: The request to be sent" }
+@Param { value:"soapVersion: The soap version of the request" }
+@Return { value:"http:Request: Returns the soap Request as http:Request with the soap envelope" }
+function fillSOAPEnvelope (Request request, SoapVersion soapVersion) (http:Request) {
+    xml soapEnv = createSoapEnvelop(soapVersion);
+    xml soapPayload = createSoapHeader(request, soapVersion);
+    if (request.payload != null) {
+        xml body = createSoapBody(request.payload, soapVersion);
+        soapPayload = soapPayload + body;
+    }
+    soapEnv.setChildren(soapPayload);
+    http:Request req = {};
+
+    req.setXmlPayload(soapEnv);
+    if (soapVersion == SoapVersion.SOAP11) {
+        req.setHeader("Content-Type", "text/xml");
+        req.addHeader("SOAPAction", request.soapAction);
+    } else {
```

Comment:
Shouldn't else block be in a separate line?

Reply:
Is it a convention? If so can you please point me to that?

Reply:
According to the latest formatting rules it seems not required.